lib/src/test/java/io/ably/lib/http/HttpHeaderTest.java

```diff
@@ -0,0 +1,197 @@
+package io.ably.lib.http;
+
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Map;
+
+import fi.iki.elonen.NanoHTTPD;
+import fi.iki.elonen.router.RouterNanoHTTPD;
+import io.ably.lib.rest.AblyRest;
+import io.ably.lib.test.common.Helpers;
+import io.ably.lib.test.common.Setup;
+import io.ably.lib.types.AblyException;
+import io.ably.lib.types.ClientOptions;
+import io.ably.lib.types.Param;
+
+import static org.mockito.Mockito.mock;
+
+/**
+ * Created by VOstopolets on 8/17/16.
+ */
+public class HttpHeaderTest {
+
+    private static SessionHandlerNanoHTTPD server;
+
+    @BeforeClass
+    public static void setUp() throws IOException {
+        // create custom RouterNanoHTTPD class for getting session object
+        server = new SessionHandlerNanoHTTPD(27331);
+        server.start(NanoHTTPD.SOCKET_READ_TIMEOUT, true);
+
+        while (!server.wasStarted()) {
```

Comment:
What do you mean? I get this code from existing Http.class https://github.com/ably/ably-java/blob/master/lib/src/test/java/io/ably/lib/http/HttpTest.java#L46-L64